proteus/mprans/SpatialTools.py

```diff
@@ -86,6 +88,60 @@ def _attachAuxiliaryVariable(self, key):
                 self.auxiliaryVariables[key] = True
             if key == 'RelaxZones':
                 self.auxiliaryVariables[key] = self.zones
+            if str(key).startswith('Gauge_'):
+                self.auxiliaryVariables[key] = [gauge]
+        elif str(key).startswith('Gauge_') \
+                and gauge not in self.auxiliaryVariables[key]:
+            self.auxiliaryVariables[key] += [gauge]
```

Comment:
Just to check your indentation, you mean:
```python
if key not in self.auxiliaryVariables:
    if key == 'RigidBody':
        self.auxiliaryVariables[key] = True
    elif key == 'RelaxZones':
        self.auxiliaryVariables[key] = self.zones
    elif str(key).startswith('Gauge_'):
        self.auxiliaryVariables[key] = [gauge]
    else:
        logEvent("Was not able to append %s  to auxiliaryVariables".format(str(key)))
elif str(key).startswith('Gauge_') \
     and gauge not in self.auxiliaryVariables[key]:
    self.auxiliaryVariables[key] += [gauge]
else:
    logEvent("Could not....")
```
correct?  I think that's a good compromise between an error and invisible responses, although I'm not sure how high level the logEvent should be (it depends on how much it gets spammed by repeated calls to the function - if it's a lot, it maybe should be a higher log level).

I'd also put in a third warning (or at least break up the logic to avoid having the second warning cover as much of the error-space):
```python
if key not in self.auxiliaryVariables:
    if key == 'RigidBody':
        self.auxiliaryVariables[key] = True
    elif key == 'RelaxZones':
        self.auxiliaryVariables[key] = self.zones
    elif str(key).startswith('Gauge_'):
        self.auxiliaryVariables[key] = [gauge]
    else:
        logEvent("Phrase One")
elif str(key).startswith('Gauge_'):
    if gauge not in self.auxiliaryVariables[key]:
        self.auxiliaryVariables[key] += [gauge]
    else:
        logEvent("Phrase Two")
else:
    logEvent("Phrase Three")
```

I'd suggest for **Phrase One**:
```python
logEvent("auxiliaryVariable key: {key} not recognized.".format(key=str(key)))
```
and for **Phrase Two:**
```python
logEvent("Attempted to put identical gauge at key: {key}".format(key=str(key)))
```
(although this one may be less necessary, since its repetition is more clearly undesirable)

and finally for **Phrase Three**:
```python
logEvent("Key {key} is already attached.".format(key=str(key)))
```